core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -105,6 +109,39 @@ open class RestActionTestCaseNamingStrategy(
         return if (parentPathQualifier == "/") "" else getPath(parentPathQualifier)
     }
 
+    /*
+     * If any test uses query prams, then add withQueryParam(s) after the path to them to make them differ.
+     * The filter call ensures that we are only performing this disambiguation when there's only one individual that
+     * differs and the list of query params is not empty.
+     */
+    private fun checkForQueryParams(duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>): Map<EvaluatedIndividual<*>, String> {
+        return duplicatedIndividuals
+            .groupBy {
+                (it.evaluatedMainActions().last().action as RestCallAction).parameters.filterIsInstance<QueryParam>()
+            }
+            .filter { it.value.size == 1 && it.key.isNotEmpty()}
+            .mapNotNull { entry ->
+                val eInd = entry.value[0]
+                duplicatedIndividuals.remove(eInd)
```

Comment:
why this? this is a side effect on the input list. quite confusing, considering that the name of the function is `check`. also, this is the same datastructure used on line 118 at the beginning of stream operations (althugh these ones are eagerly evaluated), which complicates the understanding of the function. this needs refactoring/simplification

Reply:
as rule of thumb, side effects on inputs data structures should be avoided, unless explicitely clarified in the method name (somehow)

Reply:
Just in case, it follows the same pattern as the path disambiguation in the previous [PR](https://github.com/WebFuzzing/EvoMaster/pull/1121/files#diff-1635090843debc863feecb95a31a82cbb08c67fd5b7a66b03582062b01405d8dR77). It's also clarified in the [Javadoc](https://github.com/WebFuzzing/EvoMaster/pull/1127/files#diff-1635090843debc863feecb95a31a82cbb08c67fd5b7a66b03582062b01405d8dR51-R52) of the top method.

Still, I'll work on refactor, simplification and naming

Reply:
I've just pushed a draft on a possible solution. It transfers the responsibility of removing the solved duplicates to a specific function.
The downside is the duplicate removal happens twice:
1. in the `resolveAmbiguities` function where the local working copy has the resolved duplicates removed
2. in the outer loop which triggers the resolution in [NumberedTestCaseNamingStrategy](https://github.com/WebFuzzing/EvoMaster/pull/1127/files#diff-15b9fc3f018892c963e1bdab6cf898d1e32cfe9a4c3755b4249a025beca324cbR55-R57).

However, this downside could be an advantage if we decide that all ambiguity solvers should be executed for resolution. It could even be tested for the experimentation: "do you like this naming where it only solves by one particular parameter or do you want the whole enchilada?".

Let me know what you think. Once we decide on this I'll also extract the `groupBy` sections of the solvers to more clearly named methods